omegaconf/omegaconf.py

```diff
@@ -483,8 +485,22 @@ def register_new_resolver(
             name not in BaseContainer._resolvers
         ), "resolver {} is already registered".format(name)
 
+        sig = inspect.signature(resolver)
+        pass_parent = "_parent_" in sig.parameters
+        pass_root = "_root_" in sig.parameters
+        if pass_parent and use_cache:
+            raise ValueError(
+                "use_cache=True is incompatible with functions that receive the _node_"
```

Comment:
```suggestion
                "use_cache=True is incompatible with functions that receive the _parent_"
```

Reply:
To confuse the enemy, shhhh, the enemy is listening!